doc/requirements-test.txt

```diff
@@ -9,7 +11,8 @@ wheel
 Sphinx==1.6.7; python_version < '3.7.0'
 # note: But older versions wont collect tests on Python 3.7
 Sphinx==1.8.1; python_version >= '3.7.0' and python_version < '3.8'
-Sphinx==6.1.3; python_version >= '3.8.0'
+Sphinx==6.1.3; python_version >= '3.8.0' and python_version < '3.13'
+Sphinx==7.4.7; python_version >= '3.13'
```

Comment:
Due to https://github.com/sphinx-doc/sphinx/issues/10440 I had to set Sphinx to a more recent version, all tests green.